test/integration-tests/Sudoswap.test.ts

```diff
@@ -65,7 +64,53 @@ describe('Sudoswap', () => {
       await expect((await sudolets.ownerOf(93)).toLowerCase()).to.eq(ALICE_ADDRESS)
       // Expect that alice's account has 0.073 (plus gas) less ETH in it
       await expect(aliceBalance.sub(await ethers.provider.getBalance(alice.address))).to.eq(
-        value.add(receipt.gasUsed.mul(receipt.effectiveGasPrice))
+        price.add(receipt.gasUsed.mul(receipt.effectiveGasPrice))
+      )
+    })
+  })
+
+  // In this test we will sell token id 80 of Sudolets (0xfa9937555dc20a020a161232de4d2b109c62aa9c),
+  describe('Sell a sudolet on sudoswap', () => {
+    // to sell we must first buy one - send 1 eth and receive change
+    beforeEach(async () => {
+      const oneEther = parseEther('1')
+
+      const calldata = SUDOSWAP_INTERFACE.encodeFunctionData('robustSwapETHForSpecificNFTs', [
+        [[[SUDOLETS_PAIR, ['80']], oneEther]],
+        ALICE_ADDRESS,
+        ALICE_ADDRESS,
+        1665685098,
+      ])
+      planner.addCommand(CommandType.SUDOSWAP, [oneEther, calldata])
+      planner.addCommand(CommandType.SWEEP, [ETH_ADDRESS, MSG_SENDER, 0])
+      const { commands, inputs } = planner
+
+      await (await router['execute(bytes,bytes[],uint256)'](commands, inputs, DEADLINE, { value: oneEther })).wait()
+
+      planner = new RoutePlanner()
+    })
+
+    it('sell token ids 80', async () => {
+      // put NFT in the router TODO replace with Permit2 721
+      await sudolets.transferFrom(alice.address, router.address, 80)
+      await expect((await sudolets.ownerOf(80)).toLowerCase()).to.eq(router.address.toLowerCase())
+
+      const calldata = SUDOSWAP_INTERFACE.encodeFunctionData('robustSwapNFTsForToken', [
+        [[[SUDOLETS_PAIR, ['80']], '0']],
+        ALICE_ADDRESS,
+        DEADLINE,
+      ])
+      const ethReceived = BigNumber.from('18046875000000002')
+
+      planner.addCommand(CommandType.SUDOSWAP_SELL, [calldata, SUDOLETS_721, SUDOLETS_ROUTER, 80])
+      const { commands, inputs } = planner
+
+      const aliceBalance = await ethers.provider.getBalance(alice.address)
+      const receipt = await (await router['execute(bytes,bytes[],uint256)'](commands, inputs, DEADLINE)).wait()
+
+      await expect((await sudolets.ownerOf(80)).toLowerCase()).to.eq(SUDOLETS_PAIR)
+      await expect((await ethers.provider.getBalance(alice.address)).sub(aliceBalance)).to.eq(
+        ethReceived.sub(receipt.gasUsed.mul(receipt.effectiveGasPrice))
```

Comment:
this is more of just a personal preference thing bit, but I like to make constants so that my expect lines read super clearly. Looking back on this linked line though I think I would've been a bit more explicit i.e.

expect(ethBefore.sub(ethAfter).add(gasSpent)).to.eq(nftPrice)
orrr maybe??
expect(ethDelta.sub(gasSpent)).to.eq(nftPrice)

https://github.com/Uniswap/universal-router/blob/main/test/integration-tests/Seaport.test.ts#L117-L121